### ISSUE TITLE
False negative for ``modified-iterating-list`` when using the ``del`` keyword

### ISSUE THREAD

**Issue by Pierre-Sassoulas** [PyCQA/pylint#6648]
### Bug description

```python
letters=['a','b','c','d','e','f','g','h','i','j','k','l']
for i in letters:
    del i
print(letters)  # Prints ['b', 'd', 'f', 'h', 'j', 'l']
```

### Command used

```shell
pylint a.py
```

### Pylint output

```shell
Nothing.
```

### Expected behavior

``modified-iterating-list`` raised. See https://github.com/PyCQA/pylint/issues/2471#issuecomment-1130962298

### Pylint version

```shell
2.14.0
```
